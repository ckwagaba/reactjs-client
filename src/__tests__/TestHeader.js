import React from 'react';
import { shallow } from 'enzyme';
import Header from '../components/Header';

describe('Header component', () => {

  const component = shallow(
    <Header />
  );

  it('should exist', () => {
    expect(component).toMatchSnapshot();
  });

});
