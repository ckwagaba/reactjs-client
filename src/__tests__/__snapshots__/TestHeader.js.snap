// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should exist 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="header_left"
>
          <Link
                    replace={false}
                    to={undefined}
          >
                    <div
                              className="round_button create_button"
                    >
                              +
                    </div>
          </Link>
          <span />
          <span
                    className="bucketlist_name"
          />
</div>,
        <div
          className="search"
>
          <SearchBar
                    handleSearch={undefined}
                    placeholder="Enter Search Term"
                    searchTerm={undefined}
          />
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Link
              replace={false}
              to={undefined}
>
              <div
                            className="round_button create_button"
              >
                            +
              </div>
</Link>,
            <span />,
            <span
              className="bucketlist_name"
/>,
          ],
          "className": "header_left",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <div
                className="round_button create_button"
>
                +
</div>,
              "replace": false,
              "to": undefined,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "+",
                "className": "round_button create_button",
              },
              "ref": null,
              "rendered": "+",
              "type": "div",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "bucketlist_name",
            },
            "ref": null,
            "rendered": null,
            "type": "span",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <SearchBar
            handleSearch={undefined}
            placeholder="Enter Search Term"
            searchTerm={undefined}
/>,
          "className": "search",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "handleSearch": undefined,
            "placeholder": "Enter Search Term",
            "searchTerm": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="header_left"
>
            <Link
                        replace={false}
                        to={undefined}
            >
                        <div
                                    className="round_button create_button"
                        >
                                    +
                        </div>
            </Link>
            <span />
            <span
                        className="bucketlist_name"
            />
</div>,
          <div
            className="search"
>
            <SearchBar
                        handleSearch={undefined}
                        placeholder="Enter Search Term"
                        searchTerm={undefined}
            />
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Link
                replace={false}
                to={undefined}
>
                <div
                                className="round_button create_button"
                >
                                +
                </div>
</Link>,
              <span />,
              <span
                className="bucketlist_name"
/>,
            ],
            "className": "header_left",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <div
                  className="round_button create_button"
>
                  +
</div>,
                "replace": false,
                "to": undefined,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "+",
                  "className": "round_button create_button",
                },
                "ref": null,
                "rendered": "+",
                "type": "div",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "bucketlist_name",
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <SearchBar
              handleSearch={undefined}
              placeholder="Enter Search Term"
              searchTerm={undefined}
/>,
            "className": "search",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "handleSearch": undefined,
              "placeholder": "Enter Search Term",
              "searchTerm": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
